src/escalus_fresh.erl

```diff
@@ -101,26 +102,51 @@ create_fresh_user(Config, UserName) when is_atom(UserName) ->
 
 %%% Stateful API
 %%% Required if we expect to be able to clean up autogenerated users.
-start(_Config) -> ensure_table_present(nasty_global_table()).
-stop(_) -> nasty_global_table() ! bye.
+start(_Config) ->
+    wpool:start(),
+    ensure_table_present(nasty_global_table()).
+stop(_) ->
+    wpool:stop(),
+    nasty_global_table() ! bye.
 clean() ->
-    true = lists:all(fun(A) -> A == ok end,
-                     pmap(fun delete_users/1,
-                          ets:tab2list(nasty_global_table()))),
+    wpool:start_sup_pool(unregister_pool, [{workers, 10},
+                                           {overrun_warning, 3000}]),
+    L = tag(ets:tab2list(nasty_global_table())),
+    [wpool:cast(unregister_pool,
+                {?MODULE, work_on_deleting_users, [Ord, Item, self()]}),
+     || {Ord, Item} <- L],
+    ok = collect(L),
     ets:delete_all_objects(nasty_global_table()),
     ok.
 
+collect([]) ->
+    ok;
+collect(OrdItems) ->
+    receive
+        {done, Id} ->
+            collect(lists:filter(fun({Ord, _Item}) -> Ord =/= Id end, OrdItems))
+    after 5000 ->
+              error({timeout_when_unregistering,
+                     {amount, length(OrdItems)},
+                     {unregistered_items, untag(OrdItems)}})
+    end.
+
 %%% Internals
 nasty_global_table() -> escalus_fresh_db.
 
-delete_users({_Suffix, Conf}) ->
-    Plist = proplists:get_value(escalus_users, Conf),
-    escalus_users:delete_users(Conf, Plist),
+work_on_deleting_users(Ord, {_Suffix, Conf} = Item, CollectingPid) ->
+    do_delete_users(Conf),
+    CollectingPid ! {done, Ord},
```

Comment:
if do_delete_users crashes, than our client would not receive result and we would wait the whole timeout.

Reply:
I agree with @arcusfelis. It'd be good to send error details if `do_delete_users` crashes.